aten/src/ATen/native/cuda/Normalization.cu

```diff
@@ -432,10 +466,15 @@ std::tuple<Tensor, Tensor, Tensor> batch_norm_backward_cuda_template(const Tenso
 
   using accscalar_t = at::acc_type<scalar_t, true>;
   Tensor grad_input_;
+  Tensor grad_input_cont;
   Tensor grad_weight_;
   Tensor grad_bias_;
+  auto input_cont = input_.reshape({input_.size(0), input_.size(1), -1});
```

Comment:
you probably want to `.contiguous()` this before `reshape`. `reshape` doesn't guarantee contiguity.

however, I am a bit puzzled here. in theory, shouldn't `run kernel with noncontig data` always be at least as fast as `make contiguous + run kernel with contig data`?

Reply:
I deliberately left it out, because I figured that I'd avoid the copy and the kernels don't need contiguity (except in the dimensions that get merged).
Regarding efficiency, I think it depends on whether it's possible to have a more efficient access pattern when making things contiguous than on non-contiguous data. In the end I figured that I might as well be going with non-contiguous data.
But I should have renamed the Tensors and will do now.